dev/flipbook.tsx

```diff
@@ -53,7 +55,22 @@ export function Flipbook() {
     const numQuestions = selectNumQuestions(state);
     const index = selectCurrentQuestionIndex(state);
 
-    const noTextEntered = state.questions.trim() === "";
+    const questionsState = state.questions.trim();
+    const noTextEntered = questionsState === "";
+
+    useEffect(() => {
+        const localStorageQuestions = localStorage.getItem(LS_QUESTIONS_KEY);
+        if (noTextEntered && localStorageQuestions) {
+            dispatch(setQuestions(localStorageQuestions));
+        }
```

Comment:
I wonder if we could simplify this `useEffect` callback, and get this logic under test, by unconditionally dispatching an event like `loadQuestionsFromStorage` and letting the reducer handle it in different ways based on the state of the question input. Something like this:

```ts
useEffect(() => {
  const storedQuestions = localStorage.getItem(LS_QUESTIONS_KEY);
  dispatch(loadQuestionsFromStorage(storedQuestions));
}, [])

useEffect(() => {
  localStorage.setItem(LS_QUESTIONS_KEY, questionsState);
}, [questionsState])
```

Reply:
Updated